crates/graphql/src/main.rs

```diff
@@ -39,19 +41,22 @@ async fn graphql(
     st: web::Data<Arc<Schema>>,
     data: web::Json<GraphQLRequest>,
 ) -> Result<HttpResponse, Error> {
-    let json = web::block(move || {
-        let res = data.execute_sync(&st, &());
-        serde_json::to_string(&res)
-    })
-    .await?;
+    let db_pool =
+    db::connect(db::ConnectMode::Read).context("Failed to connect to Postgres").unwrap();
+
+    let db_connection = db_pool.get().unwrap();
```

Comment:
use closure to pass in db_pool. reference db_connection from context in queries instead of pulling form instance of query root.